backend/internal/flow/flowexec/service.go

```diff
@@ -36,6 +36,7 @@ import (
 type FlowExecServiceInterface interface {
 	Execute(appID, flowID, actionID, flowType string, inputData map[string]string) (
 		*model.FlowStep, *serviceerror.ServiceError)
+	InitiateFlow(initContext *model.FlowInitContext) (string, *serviceerror.ServiceError)
```

Comment:
Having a separate function could be misleading for a outside component which uses this flow execution service right?
In a normal flow execution, we should call `Execute()` function to both initialize and continue an existing flow. However with `InitiateFlow()`, someone could confuse which function to call when calling the service. Ideally this should only be called from an internal component.

Reply:
As discussed offline, we will be refactoring later